### PR TITLE
Add getStrategyName() method to report strategy names

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

**Review by mcfadden8 — APPROVED** [review]
Looks good!  Would it be worth adding a cookbook recipe for this?

**Comment by davidbeckingsale** [comment]
> Looks good! Would it be worth adding a cookbook recipe for this?

Added.

**Review by mcfadden8 — APPROVED** [review]

File: examples/cookbook/recipe_strategy_name.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include <iostream>
+
+#include "umpire/Allocator.hpp"
+#include "umpire/ResourceManager.hpp"
+#include "umpire/strategy/QuickPool.hpp"
```
Comment (kab163) [inline examples/cookbook/recipe_strategy_name.cpp:11]: Some of these includes aren't needed I don't think - at least not QuickPool

Comment (davidbeckingsale) [reply examples/cookbook/recipe_strategy_name.cpp:11]: Fixed. I meant to use QuickPool.

**Review by kab163 — APPROVED** [review]

File: examples/cookbook/recipe_strategy_name.cpp (hunk 1 above)
Comment (kab163) [reply examples/cookbook/recipe_strategy_name.cpp:11]: Ah I see lol

**Pull request merged**